src/main/java/cz/codecamp/logger/loggers/BaseLogger.java

```diff
@@ -0,0 +1,95 @@
+package cz.codecamp.logger.loggers;
+
+import cz.codecamp.logger.FormatterInterface;
+import cz.codecamp.logger.LogLevelEnum;
+import cz.codecamp.logger.LoggerInterface;
+import cz.codecamp.logger.formatters.SimpleFormatter;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Created by micha on 05.10.2016.
+ */
+public abstract class BaseLogger implements LoggerInterface {
+    public static Function<StackTraceElement[], StackTraceElement> DEFAULT_STACKTRACE_ELEMENT_SUPPLIER = defaultStackTraceSupplier();
+    private FormatterInterface formatter;
+    private LogLevelEnum minLogLevel;
+    private Supplier<Long> timeSupplier;
+    private Function<StackTraceElement[], StackTraceElement> stackTraceElementSupplier;
+
+    public BaseLogger() {
+        this.formatter = new SimpleFormatter();
+        this.minLogLevel = LogLevelEnum.DEBUG;
```

Comment:
Vhodnější než vytvářet `abstract` base class by imo bylo vytvořit decorator (https://en.wikipedia.org/wiki/Decorator_pattern), který odfiltruje nepotřebné log levely.